### PR TITLE
Add de.flose.Kochbuch

### PR DESCRIPTION
**Pull request by AsamK**
### Please confirm your submission meets all the criteria

Please describe your application briefly below.

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I have [built][build] and tested the submission locally.
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

This desktop application allows users to store their own recipes including pictures and categories. It provides and additional paid service to synchronise the recipes with other PCs or Android phones.

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

### CODE REVIEW

**Comment by hfiguiere** [comment]
This is missing:

> Please describe your application briefly below.

**Comment by AsamK** [comment]
> This is missing:
> 
> > Please describe your application briefly below.

Thanks, added, I missed that line.

**Comment by bbhtt** [comment]
bot, build de.flose.Kochbuch

File: de.flose.Kochbuch.json [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+                    "sha256": "fc1617a3bac38ff68ef26b0a1d46838dd341716626099b49acac663b13b55786"
+                },
+                {
+                    "type": "file",
+                    "url": "https://www.flose.de/show/kochbuch/icon.svg",
+                    "sha256": "bb2d36d15ff642cc7d9cdf41d171ff4b78e512802e5fb4058f0322e66b0b8b85"
+                },
+                {
+                    "type": "file",
+                    "url": "https://www.flose.de/download/Microsoft.VisualBasic.dll",
```
Comment (hfiguiere) [inline de.flose.Kochbuch.json:87]: what's the license for this an why is it downloaded from a third party based on the name....

File: de.flose.Kochbuch.json [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    "id": "de.flose.Kochbuch",
+    "runtime": "org.freedesktop.Platform",
+    "runtime-version": "23.08",
+    "sdk": "org.freedesktop.Sdk",
+    "sdk-extensions": [
+        "org.freedesktop.Sdk.Extension.mono6"
+    ],
+    "command": "kochbuch",
+    "finish-args": [
+        "--filesystem=home",
```
Comment (hfiguiere) [inline de.flose.Kochbuch.json:11]: why does it need this broad home filesystem access?

File: de.flose.Kochbuch.json [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+                    "url": "https://download.mono-project.com/sources/libgdiplus/libgdiplus-6.1.tar.gz",
+                    "sha256": "97d5a83d6d6d8f96c27fb7626f4ae11d3b38bc88a1726b4466aeb91451f3255b"
+                }
+            ]
+        },
+        {
+            "name": "kochbuch",
+            "buildsystem": "simple",
+            "build-commands": [
+                "install -dm755 /app/opt",
```
Comment (hfiguiere) [inline de.flose.Kochbuch.json:44]: `/app/opt` really isn't a standard path. Should just be all in `/app`

File: de.flose.Kochbuch.json (hunk 1 above)
Comment (AsamK) [reply de.flose.Kochbuch.json:87]: This file is from the Mono project using a MIT license (https://github.com/mono/mono/blob/main/LICENSE)
It's unfortunately not part of org.freedesktop.Sdk.Extension.mono6, so it needs to be downloaded separately.

File: de.flose.Kochbuch.json (hunk 2 above)
Comment (AsamK) [reply de.flose.Kochbuch.json:11]: The application allows import and export of recipe files and adding image files to recipes.
Mono winforms doesn't support portals so this is needed to allow the user to select any file in their home directory.

File: de.flose.Kochbuch.json (hunk 3 above)
Comment (AsamK) [reply de.flose.Kochbuch.json]: I've removed the `opt` folder.

File: de.flose.Kochbuch.json (hunk 1 above)
Comment (hfiguiere) [reply de.flose.Kochbuch.json:86]: I don't see it there. Maybe it should be built from Mono source? It's sounds fishy to have a random "Microsoft" file downloaded from a rand place on the internet.

Comment (AsamK) [reply de.flose.Kochbuch.json]: I've added a mono-basic module to compile the library from source.

**Comment by bbhtt** [comment]
bot, build de.flose.Kochbuch

**Pull request closed without merging**